calcom/cal.com
# perf: use getSlimServerSession for session retrieval

## What does this PR do?

Creates a new `getSlimServerSession` method and replaces all calls to `getServerSession`.

The new method is much faster than `getServerSession` with it not requiring the NextAuth options object which includes a number of additional packages.

Additionally introduces a primitive in-memory cache for slim sessions. This cache may need to be reworked prior to this PR being accepted since it could affect self-hosters due to lack of TTL and such, if we feel strongly about that we could use something like `lru-cache` instead which would automatically clear out less frequently accessed entries.

I've documented some caveats to this new method within the file so please have a read through.

## Type of change

- Performance Improvement

## How should this be tested?

- Verify that you are still able to access all endpoints as per normal.
- Ideally verify that the speed is faster but that's harder to do without tracing.
